### ISSUE TITLE
Wrap up work for releasing new FluentUI React Icons

### ISSUE THREAD

**Issue by tomi-msft** [microsoft/fluentui#16738]
### Describe the issue:

- [x] Port IconProvider to react-icons package once its published to add override capability
    - Overriding will be available for `react-icons-mdl2` using `IconProvider`. Need to decide on best way to provide this functionality in new package.
- [x] Create migration guide to help customers who are using font-icons to transition to new svg icons
     - [Migration Guide](https://hackmd.io/RPZl3VIDRz-sobbE48IJuQ?view) is created. Will need to add more as the work progresses in incorporating the new icons into our repo and partner's repo

**Comment by JustSlone** [comment]
Update:
- Started on migration guide (based on OWA and Fluid work)
- Still waiting on react-icons package to be published

Goal today:
- Finish doc updates PRs so we can release open source mdl2 set of icons

**Comment by tomi-msft** [comment]
Update:

- Migration guide still in progress, currently in a [hackmd](https://hackmd.io/RPZl3VIDRz-sobbE48IJuQ)
   - Sync with Jason Custer to add design specs for more context?
- [PR](https://github.com/microsoft/fluentui-system-icons/pull/182) for react-icons package out.
  - Finalize react-icon implementation, address comments on props, and publish

**Comment by tomi-msft** [comment]
Update:

    - [react-icons](https://github.com/microsoft/fluentui-system-icons/pull/182) package has been published
       - Accessible through `@fluentui/react-icons`
   - [HackMD](https://hackmd.io/RPZl3VIDRz-sobbE48IJuQ) for the migration guide available
      - Get feedback from partners about what they need to migrate

To Do:
  - Incorporate new icons to the repo
  - Incorporate new icons into a partner's repo

**Comment by tomi-msft** [comment]
Update:

- Close this issue and open a separate issue for:
    - Incorporate new icons to fluentui repo
    - Incorporate new icons to partners repo
    - Use the learnings to add to the Migration guide
    - Add Migration guide to package and update icon website
